game-core/src/test/java/games/strategy/triplea/delegate/UnitComparatorTest.java

```diff
@@ -0,0 +1,57 @@
+package games.strategy.triplea.delegate;
+
+import static games.strategy.triplea.delegate.GameDataTestUtil.armour;
+import static games.strategy.triplea.delegate.GameDataTestUtil.germans;
+import static games.strategy.triplea.delegate.GameDataTestUtil.getDelegateBridge;
+import static games.strategy.triplea.delegate.GameDataTestUtil.load;
+import static games.strategy.triplea.delegate.GameDataTestUtil.moveDelegate;
+import static games.strategy.triplea.delegate.GameDataTestUtil.territory;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+import games.strategy.engine.data.GameData;
+import games.strategy.engine.data.ITestDelegateBridge;
+import games.strategy.engine.data.PlayerID;
+import games.strategy.engine.data.Route;
+import games.strategy.engine.data.Territory;
+import games.strategy.engine.data.Unit;
+import games.strategy.triplea.xml.TestMapGameData;
+
+public final class UnitComparatorTest {
+  private static void startCombatMoveFor(final PlayerID playerId, final GameData gameData) {
+    final MoveDelegate moveDelegate = moveDelegate(gameData);
+    final ITestDelegateBridge bridge = getDelegateBridge(playerId, gameData);
```

Comment:
I think the problem is roughly here, this is actually not a unit test. It's a problem with most of the tripleA testing. The way to fix is likely to stop trying to create custom game states, but instead rework modules to be *told* what game state they are in (IOC principle)